mp2-v1/tests/common/cases/query.rs

```diff
@@ -896,41 +895,61 @@ pub async fn prove_non_existence_row<'a>(
             .table
             .execute_row_query(&query.unwrap(), &[])
             .await?;
+
         if rows.len() == 0 {
             // no node found, return None
             info!("Search node for non-existence circuit: no node found");
             return Ok(None);
         }
+
         let row_key = rows[0]
             .get::<_, Option<Vec<u8>>>(0)
             .map(RowTreeKey::from_bytea)
             .context("unable to parse row key tree")
             .expect("");
+
         // among the nodes with the same index value of the node with `row_key`, we need to find
         // the one in the highest place in the tree, i.e., a node whose parent has not the same
         // index value
-        let (mut node_ctx, node_value) = row_tree
-            .fetch_with_context_at(&row_key, primary as Epoch)
-            .await;
-        let value = node_value.value();
-        let get_parent_data = async |node_ctx: &NodeContext<RowTreeKey>| {
-            if node_ctx.parent.is_some() {
-                let parent_key = node_ctx.parent.as_ref().unwrap();
-                let (ctx, value) = row_tree
-                    .fetch_with_context_at(parent_key, primary as Epoch)
-                    .await;
-                Some((ctx, value))
+        let primary_epoch = primary as Epoch;
+        let lineage = row_tree
+            .tree()
+            .lineage(&row_key, &row_tree.at(primary_epoch))
+            .await
+            .unwrap();
```

Comment:
But the goal is to avoid doing a linear amount of SQL calls per empty blocks. I think we should refactor the test to load all the data for all the empty blocks and then use the cache. 
Right now we're still doing one psql call per empty block. 

Or maybe it's too much ?